### PR TITLE
fix the entrySelector calculation

### LINKED ISSUE #149: Fontkit produces invalid .ttf files?
I was testing the code form the homepage with the subset and it looks like the .ttf file that is produced by it is not loaded by Chrome.

Chromes throws this error:

`OTS parsing error: incorrect entrySelector for table directory`

### LINKED ISSUE #179: RangeError('Index out of range') in the layout method
Code:
```
const fontkit = require('fontkit');
const font = fontkit.openSync('simple_slum.ttf');
const run = font.layout("t’internet");
```
Output:
```
buffer.js:977
    throw new RangeError('Index out of range');
    ^

RangeError: Index out of range
    at checkOffset (buffer.js:977:11)
    at Buffer.readInt16BE (buffer.js:1121:5)
    at DecodeStream.(anonymous function) [as readInt16BE] (/home/user/temp/node_modules/restructure/src/DecodeStream.js:38:35)
    at NumberT.decode (/home/user/temp/node_modules/restructure/src/Number.js:24:38)
    at Struct._parseFields (/home/user/temp/node_modules/restructure/src/Struct.js:53:22)
    at Struct.decode (/home/user/temp/node_modules/restructure/src/Struct.js:18:12)
    at TTFGlyph._getCBox (/home/user/temp/node_modules/fontkit/index.js:11315:28)
    at TTFGlyph._getMetrics (/home/user/temp/node_modules/fontkit/index.js:11573:21)
    at TTFGlyph.get (/home/user/temp/node_modules/fontkit/index.js:11207:19)
    at TTFGlyph.descriptor.get (/home/user/temp/node_modules/fontkit/index.js:92:23)
```

fontkit v1.7.7
nodejs v8.10.0

The font is attached.
[simple_slum.zip](https://github.com/foliojs/fontkit/files/2521576/simple_slum.zip)

### PR DESCRIPTION
**Pull request by Pomax**
The spec for the searchRange and entrySelector specifies the following constraints:

>searchRange:	(Maximum power of 2 <= numTables) x 16
>entrySelector:	Log2(maximum power of 2 <= numTables)
>rangeShift:	NumTables x 16-searchRange

But the code was computing `searchRange` as just the power index, not the actual power of two that it should be.

Fixes #149 
Fixes #179

### CODE REVIEW

File: src/tables/directory.js [hunk 1]
```diff
@@ -44,8 +44,9 @@ Directory.preEncode = function(stream) {
   this.numTables = tables.length;
   this.tables = tables;
 
-  this.searchRange = Math.floor(Math.log(this.numTables) / Math.LN2) * 16;
-  this.entrySelector = Math.floor(this.searchRange / Math.LN2);
+  let maxPowerOf2 = 2 ** Math.floor((Math.log(this.numTables) / Math.LN2));
+  this.searchRange =  maxPowerOf2 * 16;
+  this.entrySelector = Math.log(maxPowerOf2) / Math.LN2;
```
Comment (Pomax) [inline src/tables/directory.js]: L47 guaranteeds that L49 is always<sup>*</sup> integer


<sup>*) up to the font-impossible value of Math.log(2**750), at least. </sup>

**Comment by Pomax** [comment]
@devongovett ?

**Comment by Pomax** [comment]
Bump, again, because it's getting new issues file over it. Can this be landed @devongovett?

File: src/tables/directory.js (hunk 1 above)
Comment (devongovett) [reply src/tables/directory.js]: I think you may need to add a Babel plugin for the exponentiation operator

Comment (Pomax) [reply src/tables/directory.js]: I'll rewrite it to Math.pow form - eventually IE will be dead and all those can be cleaned up, until then it's not really worth asking babel to do the conversion for.

**Comment by Pomax** [comment]
@devongovett switched `**` to Math.pow and added a test to the PR to catch any future repeats of this regression

**Comment by devongovett** [comment]
Thanks @Pomax!

**Pull request merged**